### PR TITLE
Add one method get_backingstore_list into Disk

### PR DESCRIPTION
**Pull request by chunfuwen**
The method is to help get backstore list from disk, then
get relative elements from backstore list,such as source file

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
```
>>> from virttest.libvirt_xml.devices.disk import Disk
>>> xml_str = '''
... <disk device="disk" type="file">
...       <driver cache="none" name="qemu" type="qcow2" />
...       <source file="/home/storage/d/d.img" index="1" />
...       <backingStore index="2" type="file">
...         <format type="qcow2" />
...         <source file="/home/storage/d/../c/c.img" />
...         <backingStore index="3" type="file">
...           <format type="qcow2" />
...           <source file="/home/storage/d/../c/../b/b.img" />
...           <backingStore index="4" type="file">
...             <format type="qcow2" />
...             <source file="/home/storage/d/../c/../b/../a/a.img" />
...             <backingStore index="5" type="network">
...               <format type="qcow2" />
...               <source name="iqn.2021-05.com.virttest:emulated-iscsi.target/0" protocol="iscsi">
...                 <host name="127.0.0.1" port="3260" />
...               </source>
...             </backingStore>
...           </backingStore>
...         </backingStore>
...       </backingStore>
...       <mirror job="active-commit" type="file">
...         <format type="qcow2" />
...         <source file="/home/storage/d/../c/../b/b.img" index="3" />
...       </mirror>
...       <target bus="virtio" dev="vda" />
...       <alias name="virtio-disk0" />
...       <address bus="0x04" domain="0x0000" function="0x0" slot="0x00" type="pci" />
...     </disk>
... '''
>>> test_disk = Disk()
>>> test_disk.xml = xml_str
>>> backstore_lst = test_disk.get_backingstore_list()
>>> parse_source_file_list = []
>>> for elem in backstore_lst:
...     src_file = None
...     src_element = elem.find('source')
...     src_file = src_element.get('file')
...     if src_file is None:
...         src_file = src_element.get('name')
...     parse_source_file_list.append(src_file)
... 
>>> print(parse_source_file_list)
['/home/storage/d/../c/c.img', '/home/storage/d/../c/../b/b.img', '/home/storage/d/../c/../b/../a/a.img', 'iqn.2021-05.com.virttest:emulated-iscsi.target/0']
>>> 
```

File: virttest/libvirt_xml/devices/disk.py [hunk 1]
```diff
@@ -231,6 +233,17 @@ def new_backingstore(self, **dargs):
             setattr(new_one, key, value)
         return new_one
 
+    def get_backingstore_list(self):
+        """
+        Return a disk backingstore list
+        """
```
Comment (dzhengfy) [inline virttest/libvirt_xml/devices/disk.py:239]: The docstring could be updated for :return:

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: virttest/libvirt_xml/devices/disk.py (hunk 1 above)
Comment (chunfuwen) [reply virttest/libvirt_xml/devices/disk.py:239]: @dzhengfy 
updated

**Comment by chunfuwen** [comment]
new test result with element_tree from virttest:

```
from virttest import element_tree as ET
from virttest.libvirt_xml.devices.disk import Disk
from virttest.libvirt_xml.devices.disk import Disk
xml_str = '''
<disk device="disk" type="file">
      <driver cache="none" name="qemu" type="qcow2" />
      <source file="/home/storage/d/d.img" index="1" />
      <backingStore index="2" type="file">
        <format type="qcow2" />
        <source file="/home/storage/d/../c/c.img" />
        <backingStore index="3" type="file">
          <format type="qcow2" />
          <source file="/home/storage/d/../c/../b/b.img" />
          <backingStore index="4" type="file">
            <format type="qcow2" />
            <source file="/home/storage/d/../c/../b/../a/a.img" />
            <backingStore index="5" type="network">
              <format type="qcow2" />
              <source name="iqn.2021-05.com.virttest:emulated-iscsi.target/0" protocol="iscsi">
                <host name="127.0.0.1" port="3260" />
              </source>
            </backingStore>
          </backingStore>
        </backingStore>
      </backingStore>
      <mirror job="active-commit" type="file">
        <format type="qcow2" />
        <source file="/home/storage/d/../c/../b/b.img" index="3" />
      </mirror>
      <target bus="virtio" dev="vda" />
      <alias name="virtio-disk0" />
      <address bus="0x04" domain="0x0000" function="0x0" slot="0x00" type="pci" />
    </disk>
'''
test_disk = Disk()
test_disk.xml = xml_str
backingstore_list = []
tree = ET.parse(test_disk.xml)
root = tree.getroot()
for elem in root.getiterator('backingStore'):
    backingstore_list.append(elem)

parse_source_file_list = [elem.find('source').get('file') or elem.find('source').get('name') for elem in backingstore_list]

>>> parse_source_file_list
['/home/storage/d/../c/c.img', '/home/storage/d/../c/../b/b.img', '/home/storage/d/../c/../b/../a/a.img', 'iqn.2021-05.com.virttest:emulated-iscsi.target/0']
```

**Comment by chunfuwen** [comment]
@dzhengfy 
@kylazhang 
Code is updated as per discussion.

File: virttest/libvirt_xml/devices/disk.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 http://libvirt.org/formatdomain.html#elementsDisks
 """
 
+from virttest import element_tree as ET
```
Comment (chloerh) [inline virttest/libvirt_xml/devices/disk.py:7]:
```suggestion
```

File: virttest/libvirt_xml/devices/disk.py [hunk 3]
```diff
@@ -231,6 +233,17 @@ def new_backingstore(self, **dargs):
...
         return new_one
 
+    def get_backingstore_list(self):
+        """
+        :return: a disk backingstore list
+        """
+        backingstore_list = []
+        tree = ET.parse(self.xml)
+        root = tree.getroot()
+        for elem in root.getiterator('backingStore'):
```
Comment (chloerh) [inline virttest/libvirt_xml/devices/disk.py:243]:
```suggestion
        for elem in self.xmltreefile.getiterator('backingStore'):
```

File: virttest/libvirt_xml/devices/disk.py (hunk 2 above)
Comment (chunfuwen) [reply virttest/libvirt_xml/devices/disk.py]: @chloerh 
updated

File: virttest/libvirt_xml/devices/disk.py (hunk 3 above)
Comment (chunfuwen) [reply virttest/libvirt_xml/devices/disk.py]: @chloerh 
updated

**Comment by chunfuwen** [comment]
new test result with using self.xmlfile.getiterator(xx):
```
>>> from virttest.libvirt_xml.devices.disk import Disk
>>> xml_str = '''
... <disk device="disk" type="file">
...       <driver cache="none" name="qemu" type="qcow2" />
...       <source file="/home/storage/d/d.img" index="1" />
...       <backingStore index="2" type="file">
...         <format type="qcow2" />
...         <source file="/home/storage/d/../c/c.img" />
...         <backingStore index="3" type="file">
...           <format type="qcow2" />
...           <source file="/home/storage/d/../c/../b/b.img" />
...           <backingStore index="4" type="file">
...             <format type="qcow2" />
...             <source file="/home/storage/d/../c/../b/../a/a.img" />
...             <backingStore index="5" type="network">
...               <format type="qcow2" />
...               <source name="iqn.2021-05.com.virttest:emulated-iscsi.target/0" protocol="iscsi">
...                 <host name="127.0.0.1" port="3260" />
...               </source>
...             </backingStore>
...           </backingStore>
...         </backingStore>
...       </backingStore>
...       <mirror job="active-commit" type="file">
...         <format type="qcow2" />
...         <source file="/home/storage/d/../c/../b/b.img" index="3" />
...       </mirror>
...       <target bus="virtio" dev="vda" />
...       <alias name="virtio-disk0" />
...       <address bus="0x04" domain="0x0000" function="0x0" slot="0x00" type="pci" />
...     </disk>
... '''
>>> test_disk = Disk()
>>> test_disk.xml = xml_str
>>> backingstore_list = []
>>> for elem in test_disk.xmltreefile.getiterator('backingStore'):
...     backingstore_list.append(elem)
... 
>>> parse_source_file_list = [elem.find('source').get('file') or elem.find('source').get('name') for elem in backingstore_list]
>>> parse_source_file_list
['/home/storage/d/../c/c.img', '/home/storage/d/../c/../b/b.img', '/home/storage/d/../c/../b/../a/a.img', 'iqn.2021-05.com.virttest:emulated-iscsi.target/0']
>>> type(backingstore_list[0])
<class 'virttest.element_tree._ElementInterface'>
>>> 
```

**Review by kylazhang — APPROVED** [review]

File: virttest/libvirt_xml/devices/disk.py [hunk 4]
```diff
@@ -231,6 +231,22 @@ def new_backingstore(self, **dargs):
             setattr(new_one, key, value)
         return new_one
 
+    def get_backingstore_list(self):
+        """
+        Usage: get source file attribute from backingStore
+            test_disk = Disk()
+            backingstore_list = []
+            for elem in test_disk.xmltreefile.getiterator('backingStore'):
+                backingstore_list.append(elem)
```
Comment (kylazhang) [inline virttest/libvirt_xml/devices/disk.py:240]: please update them as you should using "get_backingstore_list" for your usage

Comment (chunfuwen) [reply virttest/libvirt_xml/devices/disk.py]: @kylazhang 
thanks, now updated

**Review by chloerh — APPROVED** [review]

**Comment by kylazhang** [comment]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**